### ISSUE TITLE
How to add mask supervision for object 3d reconstruction  ?  

### ISSUE THREAD

**Issue by Jiakui** [Open3DVLab/NeuRodin#2]
Dear author, 

How to add mask supervision for object 3d reconstruction  ?   

Thanks

**Comment by yyfz** [comment]
Hi, thank you for your interest in our work.

Please refer to [this link](https://github.com/Open3DVLab/NeuRodin/blob/main/nerfstudio/data/dataparsers/sdfstudio_dataparser.py#L308-L323) for guidance on how to read mask images in your dataparser. You can use this as a reference to incorporate the code snippet for reading masks into your own dataparser and set `include_foreground_mask` to True.

Additionally, in our model, you can find the original mask supervision code belonging to *sdfstudio/nerfstudio* at [this location](https://github.com/Open3DVLab/NeuRodin/blob/main/nerfstudio/models/neurodin.py#L746-L752). Simply set `fg_mask_loss_mult` to a value greater than 0.

It's worth noting that we have not yet experimented with using masks for supervision. Therefore, if possible, it would be advisable to remove the mask portion from the unbias loss at [this location](https://github.com/Open3DVLab/NeuRodin/blob/main/nerfstudio/models/neurodin.py#L828-L846), as this part of the unbias loss is meaningless.

In summary, once your dataparser is ready to read mask images and your model is prepared, you only need to make the following adjustments to your original startup script:
```bash
ns-train ... --pipeline.model.fg_mask_loss_mult <Coefficient of mask supervision> ... <your-data-parser (for example `sdfstudio-data`> ... --include_foreground_mask True
```
